lib/systems/inspect.nix

```diff
@@ -91,13 +91,18 @@ rec {
     isMusl         = with abis; map (a: { abi = a; }) [ musl musleabi musleabihf muslabin32 muslabi64 ];
     isUClibc       = with abis; map (a: { abi = a; }) [ uclibc uclibceabi uclibceabihf ];
 
-    isEfi = [
+    # This is the list of platforms which can potentially have a UEFI firmware
+    # This is helpful to know when to emit UEFI binaries along your software, e.g. for boot or more.
+    hasEfi = [
       { cpu = { family = "arm"; version = "6"; }; }
       { cpu = { family = "arm"; version = "7"; }; }
       { cpu = { family = "arm"; version = "8"; }; }
       { cpu = { family = "riscv"; }; }
       { cpu = { family = "x86"; }; }
     ];
+
+    # This name will soon be used for detection of the UEFI target itself.
+    isEfi = lib.warn "Use `platform.hasEfi` instead to detect potential EFI firmware support. `platform.isEfi` will mean 'is this an UEFI platform' in Nixpkgs 24.05." hasEfi;
```

Comment:
As I said, if we're deprecating, the deprecation warning shouldn't be implemented until _next_ release cycle, because otherwise projects wanting to support both stable and unstable would have to work around the warning.

Reply:
Apologies, right